### DISCUSSION TITLE
GitHub workflow with fugitive

### DISCUSSION THREAD

**Discussion by GordianDziwis** [tpope/vim-fugitive#1800] [Q&A] [answered]
Is it possible to do something like this bash command:

```bash
git push --set-upstream origin $(git branch --show-current)
```
with vim fugitive?

I tried:
```
:Git push --set-upstream origin system(git branch --show-current)
```
But his throws an error

```
:!git push --set-upstream origin $(git branch --show-current)
```
This did not work, because ssh auth does not work in nvim terminal.

Can somebody help me?

**Comment by tpope** [comment, marked as answer]
You can stitch this together with `:execute` but it is fraught with traps and edge cases. I recommend creating a `!` style Git alias instead.

**Reply by GordianDziwis** [reply]
I tried this, but with 

```
git push --set-upstream origin $(git branch --show-current)
```

But this returns:

```
ssh_askpass: exec(/usr/bin/ssh-askpass): No such file or directory
git@gitlab.com: Permission denied (publickey,keyboard-interactive).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

shell returned 128
```

But `:Git push` and `git push` inside my shell work.

There is this issue with the same error: https://github.com/tpope/vim-fugitive/issues/1292

`git config --list` and `:!git config --list` have an identical output

Solution was to install ssh-askpass. But this opens an ugly GUI window. Can I change that so that running `:!git ....` is consistent with `:Git ...`?

**Reply by Ditakalif** [reply]
Help me
